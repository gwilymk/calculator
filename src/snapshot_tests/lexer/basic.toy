var x = 0;
var y = 291;
var z = (x + 3) * 38299;
var negative = -3;

# this is a comment :), which has some special characters in it

print (x / y) - z;