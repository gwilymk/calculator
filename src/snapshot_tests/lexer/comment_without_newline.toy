var x = 3;

# a comment without a trailing new line