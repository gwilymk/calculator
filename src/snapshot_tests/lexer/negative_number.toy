var x = -1280;