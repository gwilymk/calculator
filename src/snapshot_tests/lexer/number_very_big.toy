var z = 239487239472491034192834701293847120;