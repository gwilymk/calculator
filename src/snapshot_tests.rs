use std::fs;

use insta::{assert_ron_snapshot, glob};

use crate::lexer::Lexer;

#[test]
fn lexer_tests() {
    glob!("snapshot_tests", "lexer/*.toy", |path| {
        let input = fs::read_to_string(path).unwrap();

        let lexer = Lexer::new(&input);
        let output = lexer
            .map(|token| match token {
                Ok((_, token, _)) => Ok(token),
                Err(e) => Err(format!("{e}")),
            })
            .collect::<Vec<_>>();

        assert_ron_snapshot!(output);
    });
}
