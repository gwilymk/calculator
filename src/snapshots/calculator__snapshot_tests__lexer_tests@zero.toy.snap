---
source: src/snapshot_tests.rs
expression: output
input_file: test_inputs/lexer/zero.toy
---
[
  Ok(KeywordVar),
  Ok(Identifier("y")),
  Ok(Assign),
  Ok(Integer("0")),
  Ok(Semicolon),
]
